CMakeLists.txt

```diff
@@ -311,6 +311,14 @@ if (ONEDPL_BACKEND MATCHES "^(tbb|dpcpp|dpcpp_only)$")
             $<$<AND:$<BOOL:${ONEDPL_USE_DEVICE_FPGA_HW}>,$<BOOL:${ONEDPL_FPGA_STATIC_REPORT}>>:-fsycl-link>
             ${ONEDPL_AOT_OPTIONS}
             )
+        
+        # if C++23 or newer, include Distributed Ranges (experimental)
+        if (CMAKE_CXX_STANDARD GREATER_EQUAL 23)
+            set(ONEDPL_USE_DR TRUE)
+            message(STATUS "Adding Distributed Ranges to the project")
+        else()
+            message(STATUS "C++23 required to use Distributed Ranges in oneDPL")
+        endif()
```

Comment:
Which specifically C++23 features are needed, and why is C++20 not sufficient? I am afraid that requiring C++23 might further limit the number of people ready to try the feature.

Reply:
This is a good point.  I will look into this and get back to you with a precise answer.  As of about 6 months ago, the only C++23 features we were using were `zip_view` and `enumerate_view`.

We do have internal versions of these that we could use (or did at some point), so I will investigate how much work it would be to revert to requiring C++20 only.

Reply:
I spent some time yesterday and today drafting a PR (#1702) to switch to C++20 only.  This essentially replaces the standard `zip_view`, `enumerate_view`, and `iota_view` with our own versions.

This *should* work, but there are still some thorny issues to work out.  In particular, it appears I wrote the `zip_view`'s iterator in a way that doesn't handle const-ness correctly---a user can interact with a const `zip_view` just fine, but certain const `zip_view`s have conflicting reference and value types.  e.g., a const value types but a non-const reference type.  This causes errors in the standard library when it looks for a [common reference](https://en.cppreference.com/w/cpp/iterator/iter_t) between them as part of the iterator concepts.  Fixing this, because of the way that `zip_view` is used internally, will likely require re-writing the `zip_view`'s iterator, which may end up being quite a bit of work.

Given the complexity of `zip_view` and that it's difficult to tell exactly how much work this is going to require, I think we should hold off on supporting C++20 for now.

Reply:
@akukanov , last few days me and @BenBrock were trying to make Distrtibuted Ranges in oneDPL compiling and working with C++20 standard. Lot of things still do not compile and most of functionality will have to be disabled if we want to lower requirements from 23 to 20 standard. We both agreed it is not good to put more work to this now. For people requiring C++20 standard and older g++ there is Distributed-Ranges repo which is based on ranges-v3 and works.

Please approve keeping C++23 requirement.

Reply:
`iota_view` is in C++20. So the problem is with just `enumerate_view` and `zip_view`, and it has two aspects: available public APIs and internal usage.

For the available public APIs, a solution is simple: limit availability of only two APIs to C++23, while the majority would work with C++20.

For the internal use, it is not that simple of course. However a good news is that there is already a "domestic" version of `zip_view` in oneDPL https://github.com/oneapi-src/oneDPL/blob/e3ebb58c1cc97261909c3bb3b47e4b3747be8dde/include/oneapi/dpl/pstl/utils_ranges.h#L146 which can be tried for internal use. Of course the necessary modifications are possible there. We do not have `enumerate_view`, but it should not be hard to make out of `zip` and `iota`.

So, let's keep looking at this problem. It would also be great if you listed the APIs that require only C++20, and those that require C++23 (including if the implementation needs `zip_view` etc.)